### PR TITLE
XFuse file overwrite and lookup improvements

### PR DESCRIPTION
**Pull request by matt335672**
This pull request fixes a couple of annoyances I've had with the xfuse filing system:-

- When connecting for the first time, file can't be viewed without browsing down to them (e.g.):-
```
$ ls -l ~/thinclient_drives/U\:/Downloads/temp.txt
ls: cannot access /home/disktest/thinclient_drives/U:/Downloads/temp.txt: No such file or directory
$ ls -l ~/thinclient_drives/U\:/Downloads
ls: cannot access /home/disktest/thinclient_drives/U:/Downloads: No such file or directory
$ ls -l ~/thinclient_drives/U\:/
<snipped>
$ ls -l ~/thinclient_drives/U\:/Downloads/
<snipped>
$ ls -l ~/thinclient_drives/U\:/Downloads/temp.txt
-rw-r--r--. 0 disktest gdat_hdv_disktest 43 Feb 28 16:08 /home/disktest/thinclient_drives/U:/Downloads/temp.txt
```
- Existing files can't be overwritten or appended (e.g.):-
```
$ echo "More text" >~/thinclient_drives/U\:/Downloads/temp.txt
bash: echo: write error: Input/output error
$ echo "More text" >>~/thinclient_drives/U\:/Downloads/temp.txt
bash: echo: write error: Input/output error
```

The first is fixed by changing the lookup functionality for remote files. The original lookup functionality only looks for file attributes in the local cache, even for remote files. Not only does this fail if the directory hasn't been read before, but long-term caching problems are introduced. This implementation always goes to the remote system to look for file attributes, relying on the fuse implementation to do any local caching.

The second of these is relatively simple; the open file functionality needs to have support added for O_WRONLY mode.

For the purpose of review and discussion, I've left this pull request as a series of four commits for now. The first two implement quite a few tidy-ups to the code, and the second two implement the features above.

I appreciate this is a fairly significant change, so I'm happy to split this up into separate pull requests if this helps.

### CODE REVIEW

**Comment by metalefty** [comment]
Thank you for your great work! Make sure to pass CI test.

**Comment by matt335672** [comment]
As an extra bit of info, I'm pretty sure, this PR addresses the issues raised in #996

**Comment by metalefty** [comment]
Let's ship this to the upcoming v0.9.10 release.

File: sesman/chansrv/devredir.c [hunk 1]
```diff
@@ -91,6 +91,13 @@
     }                                                   \
 }
 
+/*
+ * Size of structs supported by IRP_MJ_QUERY_INFORMATION without
+ * trailing RESERVED fields (MS-RDPEFS 2.2.3.3.8)
```
Comment (metalefty) [inline sesman/chansrv/devredir.c]: Thank you for recording the source (MS-RDP*)!

**Pull request merged**